flink-core/src/main/java/org/apache/flink/configuration/description/Formatter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration.description;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Allows providing multiple formatters for the description. E.g. Html formatter, Markdown formatter etc.
+ */
+public abstract class Formatter {
+
+	/**
+	 * Formats the description into a String using format specific tags.
+	 *
+	 * @param description description to be formatted
+	 * @return string representation of the description
+	 */
+	public String format(Description description) {
+		return description.getBlocks().stream().map(b -> b.format(this)).collect(Collectors.joining());
```

Comment:
the implementation here is rather weird.

First we give the formatter a description to format, which then passed itself to each block, which then again call into the formatter.
Why aren't we doing this with a simple overloaded `format` method for specific implementations, and then simply drill down through the description? (LinkElement, LineBreak, etc.)

Reply:
Originally I thought this way there will be less loops in the implementation of concrete formatters, but I had a second look and I think it is not the case. I agree with your suggestion. In other words I unnecessary overcomplicated things. :(